python/tests/src/lib/test_create_archive.py

```diff
@@ -33,19 +35,24 @@ def test_create_archive_creates_zip():
 
 
 def test_create_archive_lambda_payload():
-    data = [
-        {"organization_id": "1234", "reporting_period_id": "5678"},
-        {"organization_id": "1234", "reporting_period_id": "5678"},
-    ]
-    payload = CreateArchiveLambdaPayload.model_validate({"Payload": data})
-    assert payload.organization_id == "1234"
-    assert payload.reporting_period_id == "5678"
+    organizationObj = {
+        "organization": {
+            "id": 1234,
+            "preferences": {"current_reporting_period_id": 5678},
+        }
+    }
+    try:
+        CreateArchiveLambdaPayload.model_validate(organizationObj)
+    except ValidationError:
+        pytest.fail(
+            "CreateArchiveLambdaPayload.model_validate raised ValidationError unexpectedly"
+        )
 
 
 def test_create_archive_lambda_payload_failed():
-    data = [
-        {"organization_id": "1234", "reporting_period_id": "5678"},
-        {"organization_id": "1235", "reporting_period_id": "5678"},
-    ]
-    with pytest.raises(ValueError):
-        CreateArchiveLambdaPayload.model_validate({"Payload": data})
+    organizationObj = {
+        # "id": "1234", Missing a required field
+        "preferences": {"current_reporting_period_id": 5678},
```

Comment:
supernit: could we make this reporting period ID a global and use it across test cases since we're using the same one?

Reply:
This is the ideal I'd like to get to, but for some reason I couldn't quite get it working with how to pass parameters from step-functions to each of the lambda invocations.

Here's the ideal payload:
```
{
     common_metadata: { organization: { id: 99, preferences: { current_reporting_period: 2 } } },
     project1A: { ... details for project 1A only ... }
     project1B: { ... details for project 1B only ... }
     project1C: { ... details for project 1C only ... }
     subrecipient: { ... details for subrecipient only ... }
     zip: { ... details for zip creation only .... }
}
```

Currently I filter this in `treasury_generation_step_function.tf` [here](https://github.com/usdigitalresponse/cpf-reporter/pull/395/files#diff-a8bb03f28db1248b091274bac7a71c901996bb807c801e8233d0bb5e168f16a1R21) as follows:
`"Payload.$" : "$$.Execution.Input.['project1A']",`

According to https://jsonpath.com/ there is some union operator that can be used but I couldn't quite grok it. Lmk if anything pops out as an obvious solution here. In theory something like this would work:
`"Payload.$": "$$.Execution.Input.['project1A'], $$.Execution.Input.['common_metadata']"`